corehq/apps/consumer_user/signals.py

```diff
@@ -0,0 +1,38 @@
+from corehq.apps.consumer_user.models import ConsumerUserInvitation
+from corehq.apps.consumer_user.models import ConsumerUserCaseRelationship
+from corehq.form_processor.models import CommCareCaseSQL
+from corehq.form_processor.signals import sql_case_post_save
+from .const import CONSUMER_INVITATION_CASE_TYPE
+from .tasks import create_new_invitation
+
+
+def make_invitation_inactive(invitation, case):
+    invitation.make_inactive()
+    ConsumerUserCaseRelationship.objects.filter(case_id=case.case_id,
+                                                domain=case.domain).delete()
+
+
+def send_email_case_changed_receiver(sender, case, **kwargs):
+    if case.type != CONSUMER_INVITATION_CASE_TYPE:
```

Comment:
I think you need to add an extra check here  
```py
case.get_case_property(CONSUMER_INVITATION_STATUS) != CONSUMER_INVITATION_SENT
```
Otherwise you'll end up with an infinite loop.

Reply:
Made the change inside the method. If we keep it at top when the case is closed it wont update the corresponding invitation